docs/library/enum.rst

```diff
@@ -0,0 +1,186 @@
+    print(e)
+    print("e.is_value('B'):", e.is_value('B'))          # check if the value is in the Enum object
+    print("e.B: will raise the KeyError exception")
+    print(e.B)                                          # raise an exception due to no such a key attribute
+
+Output is::
+
+    Enum({'Y': 2, 'X': 1})
+    add new key-value pair
+    Enum({'A': 'A', 'B': 'B', 'Y': 2, 'X': 1})
+    e.X: 1
+    e.X.value: 1
+    e(e.X): 1
+    e.key_from_value(1): Enum.X
+    e.is_value(e.B): True
+    del e.B
+    Enum({'A': 'A', 'Y': 2, 'X': 1})
+    e.is_value('B'): False
+    e.B: will raise the KeyError exception
+    Traceback (most recent call last):
+    File "<stdin>", line 234, in <module>
+    File "<stdin>", line 126, in __getattr__
+    KeyError: no such attribute: B
+
+Methods
+-------
+
+.. _Enum_append:
+.. method:: Enum.append(arg, **kw_args):
+
+    Append key-value pairs with the following parameters:
+
+      - *arg* is the dictionary of key-value pairs.
+      - *kw_args* are key1=value1, key2=value2, ...
+
+.. method:: Enum.is_value(value)
+
+    Check if the value is in the Enum object.
+    Returns True if the value is in the Enum object, and False otherwise.
+
+.. method:: Enum.key_from_value(value)
+
+    Returns the key from the value. If a value is not in the Enum object, it raises an exception.
+
+Usage example::
+
+    from enum import Enum
+
+    class State(Enum):
+        Stop = 10
+        Run = 20
+        Ready = 30
+
+    state = State()
+    print("state:", State())
+
+    current_state = state.Stop
+    print("current_state:", current_state, state.key_from_value(current_state))
+    if current_state == state.Stop:
+        print(" Stop state")
+    if current_state != state.Ready:
+        print(" Not a Ready state")
+        print(" Run!")
+        current_state = state.Run
+    print("current_state:", current_state, state.key_from_value(current_state))
+    # some process
+    i = -1
+    while current_state != state.Ready:
+        i += 1
+        if state.is_value(i):
+            if state(i) == state.Ready:
+                current_state = state.Ready
+        print(".", end="")
+    print()
+    print("current_state:", current_state, state.key_from_value(current_state))
+    print("Done!")
+
+Output is::
+
+    state: State({'Ready': 30, 'Stop': 10, 'Run': 20})
+    current_state: 10 State.Stop
+    Stop state
+    Not a Ready state
+    Run!
+    current_state: 20 State.Run
+    ...............................
+    current_state: 30 State.Ready
+    Done!
+
+.. warning::
+
+    There is the trouble with the isinstance()/type().
+
+    >>> isinstance(State.Run, State)
+    False
+    >>> State(State.Ready) == State.Ready
+    False
+    >>> int(str(State(State.Ready))) == State.Ready
+    True
+    >>> type(State(State.Ready))
+    <class 'State'>
+    >>> type(state(State.Ready))
+    <class 'int'>
+    >>> state(State.Ready) == State.Ready
+    True
```

Comment:
@dpgeorge 
There is solution for this trouble.
We need to remove enums from internal `State.__dict__`
```
>>> class State(Enum):
        Stop = 10
        Run = 20
        Ready = 30

>>> dir(State)
['__dir__', '__call__', '__class__', '__delitem__', '__getattr__', '__getitem__', '__init__', '__len__', '__module__', '__name__', '__qualname__', '__repr__', '__setitem__', '__str__', 'append', 'clear', 'copy', 'get', 'items', 'keys', 'pop', 'popitem', 'setdefault', 'update', 'values', '__bases__', '__delattr__', '__dict__', '__setattr__', 'fromkeys', '_get_enums_from_class', '_update', 'is_value', 'key_from_value', 'Stop', 'Run', 'Ready']
>>> State.__dict__
{'__qualname__': 'State', 'Ready': 30, '__module__': '__main__', 'Stop': 10, 'Run': 20}
>>> 
```